src/utils/text-with-links.jsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+const rePattern = /(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w.-]+)+[\w\-._~:/?#[\]@!$&'()*+,;=.]+/gi;
+
+export default (text) => {
+  const comps = [];
+  let match = rePattern.exec(text);
+  if (!match) return <div>{text}</div>;
+  let currentIndex = 0;
+  while (match) {
+    comps.push({ type: 'text', value: text.substr(currentIndex, match.index - currentIndex) });
+    comps.push({ type: 'link', value: text.substr(match.index, match[0].length) });
+    currentIndex = match.index + match[0].length;
+
+    match = rePattern.exec(text);
+  }
+  comps.push({ type: 'text', value: text.substr(currentIndex, text.length - 1) });
+  return (
+    <div>
+      {comps.map((component, i) => {
+        const key = `${component.type}-${component.value}-${i}`;
+        return component.type === 'text' ? (
+          <span key={key}>{component.value}</span>
+        ) : (
+          <a key={key} target="_blank" href={component.value}>
+            {component.value}
+          </a>
+        );
+      })}
+    </div>
+  );
+};
```

Comment:
Because this is a jsx file I think this should be a React component: 
```
    <TextWithLinks
        text={ text }
    />
```

Maybe it should be a `span` instead of `div` too, so it can be reused without creating a `div` in tables for example

Reply:
ack

Reply:
Pure component even!